lychee-lib/src/types/preprocessor/mod.rs

```diff
@@ -0,0 +1,52 @@
+use std::{path::PathBuf, process::Command};
+
+use serde::Deserialize;
+
+use super::{ErrorKind, Result};
+
+/// Preprocess files with the specified command.
+/// So instead of reading the file contents directly,
+/// lychee will read the output of the preprocessor command.
+/// The specified command is invoked with one argument, the path to the input file.
+///
+/// For example using `cat` is equivalent to not specifying any preprocessor command.
+/// To invoke programs with custom arguments,
+/// create a shell script to specify it as preprocessor command.
+#[derive(Debug, Clone, Deserialize, PartialEq, Eq)]
+pub struct Preprocessor(String);
+
+impl From<String> for Preprocessor {
+    fn from(s: String) -> Self {
+        Self(s)
+    }
+}
+
+impl Preprocessor {
+    /// Try to invoke the preprocessor command with `path` as single argument
+    /// and return the resulting stdout.
+    pub(crate) fn process(&self, path: &PathBuf) -> Result<String> {
+        let pre = &self.0;
```

Comment:
That's kind of the reason why I prefer structs for newtypes recently. E.g. with

```rust
pub struct Preprocessor {
  command: String,
}
```

this would read nicely: `let pre = &self.command`.

I personally would also do the borrow where the command gets used, so I'd write

```rust
let command = self.command;
let output = Command::new(&command)...
```

or maybe even

```rust
let output = Command::new(&self.command)...
```


But that's definitely just a stylistic choice.